quickwit-indexing/src/actors/indexer.rs

```diff
@@ -172,19 +175,25 @@ impl IndexerState {
     fn get_or_create_indexed_split<'a>(
         &self,
         partition_id: u64,
+        last_delete_opstamp: u64,
         splits: &'a mut FnvHashMap<u64, IndexedSplit>,
         ctx: &ActorContext<Indexer>,
     ) -> anyhow::Result<&'a mut IndexedSplit> {
         match splits.entry(partition_id) {
             Entry::Occupied(indexed_split) => Ok(indexed_split.into_mut()),
             Entry::Vacant(vacant_entry) => {
-                let indexed_split = self.create_indexed_split(partition_id, ctx)?;
+                let indexed_split =
+                    self.create_indexed_split(partition_id, last_delete_opstamp, ctx)?;
                 Ok(vacant_entry.insert(indexed_split))
             }
         }
     }
 
-    fn create_workbench(&self) -> anyhow::Result<IndexingWorkbench> {
+    async fn create_workbench(&self) -> anyhow::Result<IndexingWorkbench> {
+        let last_delete_opstamp = self
+            .metastore
+            .last_delete_opstamp(&self.pipeline_id.index_id)
```

Comment:
That's one extra call every 30s per index.
Can you open an issue tracking the fact that we might want to remove this at one point?

Reply:
I did not think about it when I coded that but... we could fetch the last_delete_opstamp in the metastore directly. 
Pros are that we don't make extra call, the code is simpler on the indexer side.
Cons is that we don't have the flexibility on choosing when to fetch the last delete opstamp.

I like the solution to put that on the metastore.